src/resizer/distributors/roomsublist2.js

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2019 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import FixedDistributor from "./fixed";
+
+// const allowWhitespace = true;
+const handleHeight = 1;
+
+function log(...params) {
+    console.log.apply(console, params);
+}
+
+function clamp(height, min, max) {
+    //log(`clamping ${height} between ${min} and ${max}`);
+    if (height > max) return max;
+    if (height < min) return min;
+    return height;
+}
+
+export class Layout {
+    constructor(applyHeight, initialSizes, collapsedState) {
+        this._applyHeight = applyHeight;
+        this._sections = [];
+        this._collapsedState = collapsedState || {};
+        this._availableHeight = 0;
+        // need to store heights by id so it doesn't get
+        // assigned to wrong section when a section gets added?
+        this._sectionHeights = initialSizes || {};
+        this._originalHeights = [];
```

Comment:
the fields all need comments here, but particularly originalHeights (the heights of the sections at the point you started dragging one of them) v. heights (the heights of the sections mid-drag).